test/integration/targets/uri/tasks/use_netrc.yml

```
@@ -0,0 +1,55 @@
+- name: Write out netrc
+  copy:
+    dest: "{{ remote_tmp_dir }}/netrc"
+    content: |
+      machine httpbin.org
+      login foo
+      password bar
+
+- name: Read netrc
+  debug: var=item 
+  with_file:
+    - "{{ remote_tmp_dir }}/netrc"
+
+- name: Test Bearer authorization is expected to fail with netrc
+  uri:
+    url: https://httpbin.org/bearer
```

```suggestion
    url: https://{{ httpbin_host }}/bearer
```

The same for the remainder too.

---

`{{ httpbin_host }} `will point to `ansible.http.tests` container without the `/bearer` authentication url. 
That's why I used external httpbin.org as in original issue.
Should I keep it? or try to create the endpoint in test container?

---

If the container is missing `/bearer`, then we need to update the container image to include a newer version of httpbin.

We run into a lot of transient issues when we use `httpbin.org`, which is why we use a local test container.

---

Another option, I can use new var `{{ httpbin_org_host }}` instead of hardcoding the url. If that was the concern.

---

I just confirmed that the container does have `/bearer`

I just ran the container image (quay.io/ansible/http-test-container:2.1.0) and it responds as expected:

```
$ curl -H 'Authorization: Bearer foo' http://localhost/bearer
{"authenticated":true,"token":"foo"}
```

The default vhost (ansible.http.tests) proxies all unknown paths to httpbin which is what responds to `/bearer`

---

Thanks for confirming that. I got confused because container's Basic auth response is different from httpbin.org
Asserts and __init__ been fixed.